packages/react-jss/src/withStyles.js

```diff
@@ -39,6 +39,29 @@ const getStyles = <Theme: {}>(styles: Styles<Theme>, theme: Theme) => {
   if (typeof styles !== 'function') {
     return styles
   }
+  if (process.env.NODE_ENV !== 'production') {
+    if (styles.length < 1) {
```

Comment:
I would check for === 1, so that it includes also more than one arguments, just in case

Reply:
One generic message that covers too both cases is good enough, more than one argument is not something that many will have

Reply:
that's the second check. I separated these two cases, because 1st one is perf concern for sure, the 2nd one is misuse concern of unknown origin

Reply:
Yeah, I would make it one and write a message that covers both cases. Its not a huge problem we need to be particularly explicit about.

Reply:
"Styles function expects one `theme` argument." or something

Reply:
I would argue that warning messages should not only warn but also provide suggestions for better dx

Reply:
I don't see why cant we provide specific warning explanations for each case, because it will be removed from prod build anyway

Reply:
I think we should start adding links to the warnings with a really complete full explanation. An explanation that can't be complete is not much better than a really short one without an explanation at all in my opinion.

Reply:
For example your explanation was confusing for me.